meta-resin-common/recipes-connectivity/resin-proxy-config/resin-proxy-config/resin-proxy-config

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+#
+# copy over redsocks configuration file, set proper
+# permissions and start redsocks
+#
+
+set -e
+
+. /usr/sbin/resin-vars
+
+if [ ! -f $CONFIG_PATH ]; then
+	echo "resin-proxy-config: $CONFIG_PATH does not exist."
+	exit 1
+else
+	echo "resin-proxy-config: Found config.json in $CONFIG_PATH ."
+fi
+
+REDSOCKSCONF=${CONFIG_PATH%/*}/system-proxy/redsocks.conf
+
+if [ ! -f $REDSOCKSCONF ]; then
+	echo "resin-proxy-config: No proxy configuration found, skipping."
+	exit 0
+fi
+
+cp $REDSOCKSCONF /etc/redsocks/
+chmod 600 /etc/redsocks/redsocks.conf
+
+# Set up iptables chain for redsocks
+iptables -t nat -F REDSOCKS || true
+iptables -t nat -X REDSOCKS || true
+iptables -t nat -N REDSOCKS
+
+# Do not send to redsocks packets that are local or on the docker default bridge
+iptables -t nat -A REDSOCKS -d 0.0.0.0/8 -j RETURN
+iptables -t nat -A REDSOCKS -d 127.0.0.0/8 -j RETURN
+iptables -t nat -A REDSOCKS -d 172.17.0.1/16 -j RETURN
+iptables -t nat -A REDSOCKS -p tcp -j REDIRECT --to-ports 12345
+
+# Everything else is sent to redsocks
+iptables -t nat -A OUTPUT -p tcp -j REDSOCKS
+
+redsocks -c /etc/redsocks/redsocks.conf
```

Comment:
I did it this way to keep the redsocks recipe independent to Resin. I can do it that way, but then the redsocks package will depend on the resin-proxy-config or redsocks-prepare recipe, which has resin-specific stuff. Or have the redsocks.service inside the resin-proxy-config recipe instead?